### PR TITLE
Get a warning-free build

### PR DESCRIPTION
**Pull request by serge-sans-paille**
My first build attempt resulted in a few warnings being dumped on the console, these few patches removes them.

### CODE REVIEW

File: src/lfortran/ast_to_openmp.cpp [hunk 1]
```diff
@@ -332,7 +332,7 @@ class ASTToOPENMPVisitor : public BaseVisitor<ASTToOPENMPVisitor>
                 ) && std::string(k[1].m_id) == "len") {
                     std::swap(k[0], k[1]);
             }
-            if (x.m_type == decl_typeType::TypeCharacter &&
+            if (x.n_kind == 1 && x.m_type == decl_typeType::TypeCharacter &&
```
Comment (Pranavchiku) [inline src/lfortran/ast_to_openmp.cpp:335]: I am not sure about this change, everything else looks good.

Comment (serge-sans-paille) [reply src/lfortran/ast_to_openmp.cpp:335]: I'm not sure either. it mimics the code above though.

Comment (serge-sans-paille) [reply src/lfortran/ast_to_openmp.cpp]: ok, after second thoughts, I just dismissed this change

**Review by Pranavchiku — APPROVED** [review]
This looks good, I'll wait for one more review before merging this.

**Review by certik — APPROVED** [review]
I think that's fine, thanks!

We do turn warnings into errors at the CI, but maybe this warning only happens with newer compilers.

**Pull request merged**